app/scripts/modules/core/src/presentation/forms/fields/FormikFormField.tsx

```diff
@@ -70,13 +82,56 @@ function FormikFormFieldImpl<T = any>(props: IFormikFormFieldImplProps<T>) {
     messageNode,
   };
 
+  const [inputMode, setInputMode] = React.useState(SpelAwareInputMode.DEFAULT);
+
+  const modeToInput: { [k in SpelAwareInputMode]: React.ComponentType<IFormInputProps> } = {
+    [SpelAwareInputMode.DEFAULT]: input,
+    [SpelAwareInputMode.FREEFORM]: SimpleSpelInput,
+  };
+
+  const freeformInputEnabled = firstDefined(
+    get(props, 'spelConfig.freeformInputEnabled', null),
+    get(useContext(FormikSpelContext), 'freeformInputEnabled', null),
+    false,
+  );
+
+  React.useEffect(() => {
+    if (!freeformInputEnabled) {
+      return;
+    }
+    const fieldValue = get(props.formik.values, name, '');
+    const isFieldValueSpel = SpelService.includesSpel(fieldValue);
+    if (isFieldValueSpel) {
+      setInputMode(SpelAwareInputMode.FREEFORM);
+    }
+  }, []);
+
+  const onSpelToggleClick = () => {
+    formik.setFieldValue(name, null);
```

Comment:
I wonder if clearing the field value is the right choice.  I suspect the answer varies depending on the input.

Reply:
At least in the stage fields I played around with, it might be confusing not to clear the value, since you could end up in a state where a valid string (say, `${parameters['account']}`) is persisted on the stage model but not reflected in the UI because the user has toggled back to the default input, which is some input that only accepts some list of predefined values.

Are we comfortable clearing the field value for now as a default? What input types were you thinking it would make more sense not to clear?

Reply:
@christopherthielen are you okay merging this as-is or do you have additional thoughts on _not_ clearing the field value under certain circumstances?